src/components/role-mapping/AddRoleMappingModal.tsx

```diff
@@ -58,136 +46,71 @@ export const AddRoleMappingModal = ({
   const { t } = useTranslation("common");
   const { adminClient } = useAdminClient();
 
-  const [clients, setClients] = useState<ClientRole[]>([]);
   const [searchToggle, setSearchToggle] = useState(false);
 
-  const [selectedClients, setSelectedClients] = useState<ClientRole[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>("roles");
   const [selectedRows, setSelectedRows] = useState<Row[]>([]);
-
-  const [data, setData] = useState<Row[]>();
+  const [key, setKey] = useState(0);
+  const refresh = () => setKey(key + 1);
 
   const mapType = mapping.find((m) => m.resource === type)!;
+  const alphabetize = (rolesList: Row[]) => {
+    return sortBy(rolesList, ({ role }) => role.name?.toUpperCase());
+  };
 
-  useFetch(
-    async () => {
-      const clients = await adminClient.clients.find();
-      return (
-        await Promise.all(
-          clients.map(async (client) => {
-            const roles = await castAdminClient(
-              adminClient,
-              mapType.resource === "roles" ? "clients" : mapType.resource
-            )[mapType.functions.list[0]]({
-              id: mapType.resource === "roles" ? client.id : id,
-              clientUniqueId: client.id,
-              client: client.id,
-            });
+  const loader = async (first?: number, max?: number, search?: string) => {
+    const params: { [name: string]: string | number } = {
```

Comment:
You could use a `Record` type here.
```suggestion
    const params: Record<string, string | number> = {
```